packages/@vue/cli-service/lib/commands/serve.js

```diff
@@ -302,6 +299,20 @@ function checkInContainer () {
   }
 }
 
+function genHistoryApiFallbackRewrites (baseUrl, pages = {}) {
+  const path = require('path')
+  const multiPageRewrites = Object
+    .keys(pages)
+    .map(name => ({
+      from: new RegExp(`^/${name}`),
```

Comment:
Will fail if several page names share the same prefix, e.g. `page1`, `page10` and `page11`.

Reply:
Yes you are right. 

I found it's hardly to avoid this kind of situation. We can't simply add a trailing slash in the regex. In that way user have to add a trailing slash in url to access specified page, eg `localhost:8080/page1/`.

Any thoughts?

Reply:
What about `Object.keys(pages).sort().reverse().map(...)`?

Reply:
Nice solution, sort page names by length should be enough to avoid this issue. Thanks!